src/hooks/useMovies.tsx

```diff
@@ -0,0 +1,59 @@
+import { useState } from 'react';
+import { getMovie } from '../api';
+import { MovieData } from '../types/MovieData';
+import { Movie } from '../types/Movie';
+
+export const useMovie = () => {
+  const [preMovie, setPreMovie] = useState<Movie | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isError, setIsError] = useState<boolean>(false);
+
+  function getNormalMovie(movieFromServer: MovieData): Movie {
```

Comment:
```suggestion
  function normalizeMovie(movieFromServer: MovieData): Movie {
```
You can move this function to utils/helpers file/folder